src/pratt.rs

```diff
@@ -738,10 +759,9 @@ mod tests {
             ))
             .map(|x| x.to_string());
 
-        assert_eq!(
-            parser.parse("1+2!$*3").into_result(),
-            Ok("(((1 + (2!))$) * 3)".to_string()),
-        )
+        assert!(
+            parser.parse("§1+-~2!$*3").has_errors(),
+        );
```

Comment:
This change looks a little suspicious. Does this addition change the behaviour of existing parsers?